docs/connector-development/config-based/tutorial/5-incremental-reads.md

```diff
@@ -0,0 +1,262 @@
+# Step 5: Incremental Reads
+
+We now have a working implementation of a connector reading the latest exchange rates for a given currency.
+In this section, we'll update the source to read historical data instead of only reading the latest exchange rates.
+
+According to the API documentation, we can read the exchange rate for a specific date by querying the "/{date}" endpoint instead of "/latest".
+
+We'll now add a `start_date` property to the connector.
+
+First we'll update the spec `source_exchange_rates_tutorial/spec.yaml`
+
+```
+documentationUrl: https://docs.airbyte.io/integrations/sources/exchangeratesapi
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: exchangeratesapi.io Source Spec
+  type: object
+  required:
+    - start_date
+    - access_key
+    - base
+  additionalProperties: true
+  properties:
+    start_date:
+      type: string
+      description: Start getting data from that date.
+      pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}$
+      examples:
+        - YYYY-MM-DD
+    access_key:
+      type: string
+      description: >-
+        Your API Access Key. See <a
+        href="https://exchangeratesapi.io/documentation/">here</a>. The key is
+        case sensitive.
+      airbyte_secret: true
+    base:
+      type: string
+      description: >-
+        ISO reference currency. See <a
+        href="https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/index.en.html">here</a>.
+      examples:
+        - EUR
+        - USD
+```
+
+Then we'll set the `start_date` to last week our connection config in `secrets/config.json`.
+The following `echo` command will update your config with a start date set at 7 days prior to today.
+
+```
+echo "{\"access_key\": \"<your_access_key>\", \"start_date\": \"$(date -v -7d '+%Y-%m-%d')\", \"base\": \"USD\"}"  > secrets/config.json
+```
+
+And we'll update the `path` in the connector definition to point to `/{{ config.start_date }}`.
+Note that we are setting a default value because the `check` operation does not know the `start_date`. We'll default to hitting `/latest`:
+
+```
+retriever:
+  requester:
+      path:
+        type: "InterpolatedString"
+        string: "{{ config.start_date }}"
+        default: "/latest"
```

Comment:
Changing this section fails for me:
from 
```yaml
  retriever:
    $ref: "*ref(retriever)"
    requester:
      $ref: "*ref(requester)"
      path: "/latest"
```

to 
```yaml 
  retriever:
    requester:
        path:
          type: "InterpolatedString"
          string: "{{ config.start_date }}"
          default: "/latest"
```
Leads to the following error on `read`
```python
Exception: failed to create object of type <class 'airbyte_cdk.sources.declarative.requesters.http_requester.HttpRequester'> because __init__() missing 1 required keyword-only argument: 'url_base'`
```